src/main/java/calculator/StringAddCalculator.java

```diff
@@ -0,0 +1,55 @@
+package calculator;
+
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+class StringAddCalculator {
+
+    private static final String DEFAULT_DELIMITERS = "[,;]";
+
+    public static int run(String str) {
+        if (isEmptyOrNull(str)) {
+            return 0;
+        }
+        String[] splitStrings = split(str);
+        int[] numbers = parseStringToInt(splitStrings);
+        return sum(numbers);
+    }
+
+    private static boolean isEmptyOrNull(String str) {
+        return str == null || str.isEmpty();
+    }
+
+    private static String[] split(String str) {
+        Matcher m = Pattern.compile("//(.)\n(.*)").matcher(str);
+        if (m.find()) {
+            String customDelimiter = m.group(1);
+            return m.group(2).split(customDelimiter);
+        }
+        return str.split(DEFAULT_DELIMITERS);
+    }
+
+    private static int[] parseStringToInt(String[] strs) {
+        int[] ints = new int[strs.length];
+        for (int i = 0; i < strs.length; i++) {
+            ints[i] = parseInt(strs[i]);
+        }
+        return ints;
+    }
+
+    private static int parseInt(String stringNumber) {
+        int number = Integer.parseInt(stringNumber);
+        if (number < 0) throw new RuntimeException("음수는 허용하지 않습니다.");
+        return number;
+    }
+
+    private static int sum(int[] numbers) {
+        int sum = 0;
+        for (int number: numbers) {
+            sum += number;
+        }
+        return sum;
+    }
```

Comment:
```suggestion
    private static int sum(int[] values) {
        return Arrays.stream(values).sum();
    }
```

stream 을 활요해볼 수도 있겠네요 😉

Reply:
@Hyeon9mak
저도 스트림 정말 좋아하는데요 ㅎㅎ
코비가 이번 미션동안은 Stream을 지양하자고 하셔서 사다리타기 미션 전까지는 스트림없이 구현해보는 건 어떨까요?